doc/ssl-performance.md

```diff
@@ -0,0 +1,30 @@
+# SSL in gRPC and performance
+
+The SSL requirement of gRPC isn't necessarily making it easy to integrate. The HTTP/2 protocol requires ALPN support, which is a fairly new handshake protocol only supported by recent implementations.
+
+As a result, we've tried hard to provide a smooth experience to our users when compiling and distributing gRPC, but this may come at performance costs due to this. More specifically, we will sometime build the SSL library by disabling assembly code, which can impact performances by an order of magnitude when processing encrypted streams.
+
+Build system | Condition | Platform | Assembly code
```

Comment:
nit: Assembly code -> "Assembly code optimizations" would make the meaning of the table more self-explanatory?